### PR TITLE
[RW-6348][risk=low] Reactify ServerConfigService

### PR DESCRIPTION
**Pull request by als364**
This has many small identical changes due to the signature change when going `BehaviorSubject` -> `atom`

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [x] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/services/swagger-fetch-clients.ts [hunk 1]
```diff
@@ -87,17 +87,17 @@ function bindCtor<T extends BaseAPI>(ctor: new() => T): () => T {
 // getters for the API clients, e.g.: runtimeApi().listRuntimes();
 export const authDomainApi = bindCtor(AuthDomainApi);
 export const cdrVersionsApi = bindCtor(CdrVersionsApi);
-export const configApi = bindCtor(ConfigApi);
```
Comment (als364) [inline ui/src/app/services/swagger-fetch-clients.ts:90]: alphabetized

File: ui/src/app/utils/stores.tsx [hunk 2]
```diff
@@ -47,12 +47,6 @@ export interface CompoundRuntimeOpStore {
   [workspaceNamespace: string]: CompoundRuntimeOperation;
 }
 
-export interface StackdriverErrorReporterStore {
```
Comment (als364) [inline ui/src/app/utils/stores.tsx:50]: pulled after things that go with runtime ops

**Comment by als364** [comment]
i _think_ i have the synchronicity issues sorted at this point. running the tests locally on repeat to see if I can get a random failure.

File: ui/src/app/pages/signed-in/component.ts [hunk 3]
```diff
@@ -89,25 +93,22 @@ export class SignedInComponent implements OnInit, OnDestroy, AfterViewInit {
...
-        } else {
-          this.cdrVersionsInitialized = true;
-        }
+  async ngOnInit() {
+    // We want to block app rendering on the presence of server config
+    // data so that route components don't try to lookup config data
+    // before it's available.
+    // This will need to be a step in the React bootstrapping as well.
+    // See discussion on https://github.com/all-of-us/workbench/pull/4713
+    if (serverConfigStore.get().config) {
```
Comment (ericsong) [inline ui/src/app/pages/signed-in/component.ts:102]: is there a case that's not covered by just replacing this `if/else` block with `serverConfigStore.subscribe(() => {}`?

File: ui/src/app/services/sign-in.service.ts [hunk 4]
```diff
@@ -32,36 +31,17 @@ export class SignInService {
...
-          this.zone.run(() => {
-            this.isSignedIn.next(true);
-          });
-          return;
-        }
-      }
-
-      this.makeAuth2(config);
-    });
+    if (serverConfigStore.get().config) {
```
Comment (ericsong) [inline ui/src/app/services/sign-in.service.ts:37]: similar comment as above wrg to just using `subscribe`

**Review by ericsong — COMMENTED** [review]
looking good. I have a few nitpicky comments

File: ui/src/app/pages/signed-in/component.ts (hunk 3 above)
Comment (als364) [reply ui/src/app/pages/signed-in/component.ts:102]: yes - `serverConfigStore.subscribe` only triggers on change of `serverConfigStore`. it doesn't trigger if there is something already in the store.

File: ui/src/app/pages/admin/admin_institution.spec.tsx [hunk 5]
```diff
@@ -16,8 +17,9 @@ describe('AdminInstitutionSpec', () => {
   };
 
   beforeEach(() => {
-    serverConfigStore.next(defaultServerConfig);
+    serverConfigStore.set({config: defaultServerConfig});
 
+    registerApiClient(ConfigApi, new ConfigApiStub());
```
[the remark this answers is not in the document]
Comment (als364) [reply ui/src/app/pages/admin/admin_institution.spec.tsx:22]: good call, thanks

**Review by ericsong — APPROVED** [review]

**Review by dmohs — COMMENTED** [review]
Net -114. Very impressive!

File: ui/src/app/pages/app/component.ts [hunk 6]
```diff
@@ -40,62 +38,72 @@ export class AppComponent implements OnInit {
...
-            if (!url.match(/^https?:[/][/][a-z0-9.:-]+$/)) {
-              throw new Error('URL should be of the form "http[s]://host.example.com[:port]"');
-            }
-            this.overriddenUrl = url;
-            localStorage.setItem(LOCAL_STORAGE_API_OVERRIDE_KEY, url);
-          } else {
-            this.overriddenUrl = null;
-            localStorage.removeItem(LOCAL_STORAGE_API_OVERRIDE_KEY);
+    this.loadConfig()
+      .then(() => {
```
Comment (dmohs) [inline ui/src/app/pages/app/component.ts:51]: If you like, you may be able to reduce a level of indentation here by:
```javascript
async ngOnInit() {
  this.checkBrowserSupport()
  await this.loadConfig()
  this.loadErrorReporter()
  ...
}
```

**Pull request merged**